### ISSUE TITLE
PM2 does not install with SystemD correctly on Beaglebone Black Running Debian Jessie

### ISSUE THREAD

**Issue by Hoarfrost** [Unitech/pm2#2765]
Update 3/11/2017 3:45pm

I have edited this post in response to Unitech's post about running `pm2 startup` without the sudo command.  After his comment I created a completely new install of the OS and started from scratch.  Omitting the sudo command did improve the results, but I still needed to make a final edit manually.

#### Expected behaviour
After installing PM2 with startup command
PM2 should start after booting and start its child processes.

#### Actual behaviour
PM2 encounters error during boot process and does not start
```
debian@beaglebone:~$ journalctl -u pm2-debian.service -b
-- Logs begin at Sat 2017-03-11 20:28:46 UTC, end at Sat 2017-03-11 20:30:20 UTC. --
Mar 11 20:28:55 beaglebone systemd[1]: Starting PM2 process manager...
Mar 11 20:29:03 beaglebone systemd[1]: pm2-debian.service: Start operation timed out. Terminating.
Mar 11 20:29:03 beaglebone systemd[1]: Failed to start PM2 process manager.
Mar 11 20:29:03 beaglebone systemd[1]: pm2-debian.service: Unit entered failed state.
Mar 11 20:29:03 beaglebone systemd[1]: pm2-debian.service: Failed with result 'timeout'.
```

#### Steps to reproduce
login as debian
```
sudo npm install pm2 -g` <-- sudo required here
debian@beaglebone:~$ pm2 startup
...
[PM2] You have to run this command as root. Execute the following command:
sudo env PATH=$PATH:/usr/bin /usr/lib/node_modules/pm2/bin/pm2 startup systemd -u debian --hp /home/debian`

debian@beaglebone:~$ sudo env PATH=$PATH:/usr/bin /usr/lib/node_modules/pm2/bin/pm2 startup systemd -u debian --hp /home/debian
...
debian@beaglebone:~$ journalctl -u pm2-debian.service -b
-- Logs begin at Sat 2017-03-11 20:28:46 UTC, end at Sat 2017-03-11 20:30:20 UTC. --
Mar 11 20:28:55 beaglebone systemd[1]: Starting PM2 process manager...
Mar 11 20:29:03 beaglebone systemd[1]: pm2-debian.service: Start operation timed out. Terminating.
Mar 11 20:29:03 beaglebone systemd[1]: Failed to start PM2 process manager.
Mar 11 20:29:03 beaglebone systemd[1]: pm2-debian.service: Unit entered failed state.
Mar 11 20:29:03 beaglebone systemd[1]: pm2-debian.service: Failed with result 'timeout'.`

```

#### Steps for functional work around
It appears the problem is that the problem is the "Type=forking" in the .service file.

I was able to make PM2 function by changing _/etc/systemd/system/pm2-debian.service_

From:
```
[Service]
Type=forking 
```

To:
```
[Service]
#Type=forking 
```

#### Software versions used
```
OS         :  Linux beaglebone 4.4.49-ti-r89 #1 SMP Fri Feb 17 20:59:06 UTC 2017 armv7l GNU/Linux
node.js    : 4.8.0
PM2        : 2.4.2
```

#### PM2 Log output
Use the command: `tail --lines 50 ~/.pm2/pm2.log`
```
debian@beaglebone:~$ tail --lines 50 ~/.pm2/pm2.log
2017-03-11 20:27:50: ===============================================================================
2017-03-11 20:27:50: --- New PM2 Daemon started ----------------------------------------------------
2017-03-11 20:27:50: Time                 : Sat Mar 11 2017 20:27:50 GMT+0000 (UTC)
2017-03-11 20:27:50: PM2 version          : 2.4.2
2017-03-11 20:27:50: Node.js version      : 4.8.0
2017-03-11 20:27:50: Current arch         : arm
2017-03-11 20:27:50: PM2 home             : /home/debian/.pm2
2017-03-11 20:27:50: PM2 PID file         : /home/debian/.pm2/pm2.pid
2017-03-11 20:27:50: RPC socket file      : /home/debian/.pm2/rpc.sock
2017-03-11 20:27:50: BUS socket file      : /home/debian/.pm2/pub.sock
2017-03-11 20:27:50: Application log path : /home/debian/.pm2/logs
2017-03-11 20:27:50: Process dump file    : /home/debian/.pm2/dump.pm2
2017-03-11 20:27:50: Concurrent actions   : 2
2017-03-11 20:27:50: SIGTERM timeout      : 1600
2017-03-11 20:27:50: ===============================================================================
2017-03-11 20:28:19: ===============================================================================
2017-03-11 20:28:19: --- Stopping PM2 --------------------------------------------------------------
2017-03-11 20:28:19: Time                 : Sat Mar 11 2017 20:28:19 GMT+0000 (UTC)
2017-03-11 20:28:19: ===============================================================================
2017-03-11 20:28:19: RPC closed
2017-03-11 20:28:19: PUB closed
2017-03-11 20:28:19: PM2 successfully stopped
```

cf CONTRIBUTING.md for more informations

**Comment by Unitech** [comment]
Do not type `sudo pm2 startup`. 
Type only `pm2 startup` and pm2 will display to you the right command to copy/paste in your terminal

**Comment by Hoarfrost** [comment]
I did what Unitech suggested and used `pm2 startup` without sudo.  It worked better but there was still a problem.  I needed to manually remove the `Type=forking` from the .service file to make it work.  I have edited the original post with the new information.

**Comment by AuspeXeu** [comment]
Same problem on Raspberry pi on pm2 version 2.4.2 I downgraded to 2.4.0 and everything works as expected.

**Comment by f-hj** [comment]
Forking type is not the problem, this is a timeout. 8 seconds is too low for NodeJS on ARM, it's the same on C.H.I.P. 

Must choose higher value for ```TimeoutStartSec``` on ```/etc/systemd/system/pm2-root.service```

```
Mar 09 20:13:20 chip systemd[1]: Starting PM2 process manager...
Mar 09 20:13:28 chip systemd[1]: pm2-root.service start operation timed out. Terminating.
Mar 09 20:13:28 chip systemd[1]: Failed to start PM2 process manager.
Mar 09 20:13:28 chip systemd[1]: Unit pm2-root.service entered failed state.
```

**Comment by xbmcnut** [comment]
@f-hj Thank you! That worked for me. Had an issue with MagicMirror starting under PM2 2.4.2. Changed pm2-root.service timeout from 8 to 15s and that worked.

**Comment by vmarchaud** [comment]
As commited in https://github.com/Unitech/pm2/commit/cd4e20121507f44fde4f24c75cf581ba92bd51c1 we will remove the timeout for the next release to fix this issue.

**Comment by vmarchaud** [comment]
Published under pm2 `2.4.3` !
